### ISSUE TITLE
Unable to complete apply applying changes to citrixadc_servicegroup_servicegroupmember_binding with nameserver field present.

### ISSUE THREAD

**Issue by kevgreen-ebay-com** [citrix/terraform-provider-citrixadc#162]
Hi,

Creating this issue because we're already being assisted on it by one part of Citrix, but another part needs to do similar tests (requiring an external nameserver running on a non-privileged port) for another (off-github) issue, so this point of reference may aid co-operation. It's more or less a placeholder, or for reference of other people hitting the issue.

The object is applied on the ADC, but terraform then goes on to fail and abort the rest of apply with this error:

When applying changes to
citrixadc_servicegroup_servicegroupmember_binding.consul_pool-guvnor-nameserver,
provider "registry.terraform.io/-/citrixadc" produced an unexpected new value
for was present, but now absent.

This is where the 'nameserver' points to a vserver address on the netscaler serving on :53, which forwards to back-ends running on alternative ports, because that was the only way to make it work (there is no capability to set the nameserver to an alternate port in the binding object, which it would be nice to be able to do, also perhaps to provide multiple nameservers instead of just one).

I tried various possible workarounds in the object structure, such as tweaking dbsTTL and state, but to no avail. Only removing the 'nameserver' parameter got rid of this problem. But we require that for service discovery, so it's useless without.

It's not urgent because I've worked around it with curl.
 
K.

**Comment by giorgos-nikolopoulos** [comment]
I am unable to replicate the error you are seeing.

Could you share with us the configuration or the detailed logs of the failed operation?

For reference this is the configuration I am using

```
resource "citrixadc_servicegroup" "tf_servicegroup" {
  servicegroupname = "tf_servicegroup"
  servicetype      = "HTTP"
  autoscale        = "DNS"
}

resource "citrixadc_servicegroup_servicegroupmember_binding" "tf_binding" {
  servicegroupname = citrixadc_servicegroup.tf_servicegroup.servicegroupname
  servername       = "georgen_srv"
  port = 80
  nameserver = "10.222.74.158"
}
```

The nameserver ip `10.222.74.158` points to a DNS lb vserver bound with a functioning DNS service.

**Comment by kevgreen-ebay-com** [comment]
We're using SRV record lookup, bound through a server object. Here's the config, with some redactions:

resource "citrixadc_server" "consul_srv-myapp" {
        name = "consul_srv-myapp"
    translationip = var.consul_dns_vserver_ip
    translationmask = "255.255.255.0"
    state = "ENABLED"
    comment = "a comment. myapp"
    querytype = "SRV"
    domain = "myapp.service.consul"
}

resource "citrixadc_servicegroup" "consul_pool-myapp" {
    servicegroupname = "consul_pool-myapp"
    servicetype = "HTTP"
    autoscale = "DNS"
    cip = "DISABLED"
    usip = "NO"
    useproxyport = "YES"
    sp = "ON"
    clttimeout = "180"
    svrtimeout = "360"
    cka = "NO"
    tcpb = "NO"
    cmp = "NO"


    depends_on = [
        citrixadc_server.consul_srv-myapp,
        module.lbvserver__myapp-http,
        #module.consul-lookups. #citrixadc_lbvserver.consul-dns-mystack
    ]

    lbvservers = [ module.lbvserver__myapp-http.name ]
}

resource "citrixadc_servicegroup_servicegroupmember_binding" "consul_pool-myapp-nameserver" {
  servicegroupname = citrixadc_servicegroup.consul_pool-myapp.servicegroupname
  servername       = citrixadc_server.consul_srv-myapp.name
  dbsttl           = "1"
  nameserver       = var.consul_dns_vserver_ip
  state            = "ENABLED"

  depends_on = [
    citrixadc_servicegroup.consul_pool-myapp,
    citrixadc_server.consul_srv-myapp,
    module.lbvserver__myapp-http,
    #module.consul-lookups. #citrixadc_lbvserver.consul-dns-mystack
  ]

}

resource "citrixadc_servicegroup_lbmonitor_binding" "http-myapp-mymonitor-binding" {
    servicegroupname = citrixadc_servicegroup.consul_pool-myapp.servicegroupname
    monitorname = citrixadc_lbmonitor.mymonitor.monitorname
    weight = 80

    depends_on = [ citrixadc_servicegroup_servicegroupmember_binding.consul_pool-myapp-nameserver ]
}

# vim:sw=4:ts=4:et:

**Comment by kevgreen-ebay-com** [comment]
Looking at this again with a clear head I can see that there is nothing binding the servicegroup to the server object? So there's a missing 'servername' in the "citrixadc_servicegroup" object if I'm not mistaken?

**Comment by giorgos-nikolopoulos** [comment]
I used your configuration and still cannot replicate the error you are seeing.

Can you tell us what ADC version you are using and what kind of deployment (standalone, HA, cluster) ?
Also having the logs of the failed terraform apply run coudl be helpful.

**Comment by kevgreen-ebay-com** [comment]
It's a HA pair.

Version is "NetScaler NS13.0: Build 71.44.nc, Date: Dec 26 2020, 11:31:14   (64-bit)".

All that terraform says in non-debugging mode is given above. I'll see if I can run it again in full debug mode and send you the output privately (because of sensitive information in it etc).

**Comment by kevgreen-ebay-com** [comment]
Second time lucky, but I've emailed you privately with a debug log because it may contain sensitive information.

**Comment by kevgreen-ebay-com** [comment]
Just to add, there was an issue elsewhere with large UDP DNS responses when doing DNS service discovery, so I thought I'd check back to see whether or not this might be related, but judging by my notes and the above I think this issue happens regardless of how many backend services are returned by DNS, so I don't think it's related to that.

**Comment by giorgos-nikolopoulos** [comment]
Fixed with #278

Please use the `disable_read` option to avoid the error.
